python/solid_dmft/dmft_tools/solver.py

```diff
@@ -397,7 +397,7 @@ def solve(self, **kwargs):
 
             # Solve the impurity problem for icrsh shell
             # *************************************
-            self.triqs_solver.solve(h_int=self.h_int, **(self.solver_params | random_seed ))
+            self.triqs_solver.solve(h_int=self.h_int, **{ **self.solver_params, **random_seed })
```

Comment:
Would it be an option to just store the keyword directly in line 352 in `self.solver_params` , i.e.: 
```
self.solver_params['random_seed'] = int(self.random_seed_generator(it=kwargs["it"], rank=mpi.rank))
```

Reply:
I would rather not overwrite `solver_params['random_seed']` in-place, because that should be the value from `read_config` and if you overwrite this the wrong value might be serialized to the output file.

Reply:
But the original `solver_params` that are stored in the h5 are owned by `dmft_cylce`, and stored from there to the h5, the dict withing the solid_dmft solver object is not used for storage. Or do you mean other output? I think that should be save to overwrite then no? At least I do not see a problem right now.

Reply:
Dictionaries (and lists) are reference types. They are not copied (unless explicitly using `.copy()`):
```python
>>> a = { "foo": 1 }
>>> b = a
>>> b["foo"] = 2
>>> a
{'foo': 2}
```

Reply:
Ah yes of course. Somehow I wasn't sure with a dict but you are of course right. Then we do not want to change it ;-)